gaeguli/pipeline.c

```diff
@@ -799,6 +808,28 @@ _build_vsrc_pipeline (GaeguliPipeline * self, GError ** error)
   g_autoptr (GstPad) valve_src = NULL;
   g_autoptr (GstPluginFeature) feature = NULL;
 
+  g_autoptr (GError) gerr = NULL;
+  GPid pid;
+  gchar *arg[2];
+
+  if (g_file_test (GAEGULI_PIPELINE_WORKER, G_FILE_TEST_EXISTS)) {
+    arg[0] = GAEGULI_PIPELINE_WORKER;
+  } else {
+    arg[0] = GAEGULI_PIPELINE_WORKER_ALT;
+  }
+
+  arg[1] = NULL;
+
+  g_spawn_async (NULL, arg, NULL,
+      G_SPAWN_DO_NOT_REAP_CHILD, NULL, NULL, &pid, &gerr);
+  if (gerr != NULL) {
+    g_error ("spawning pipeline worker failed. %s", gerr->message);
+  } else {
+    g_debug ("A new child process spawned with pid %d\n", pid);
+    self->worker_pid = pid;
+    g_child_watch_add (pid, (GChildWatchFunc) _cb_child_watch, NULL);
+  }
```

Comment:
nitpick. I'd prefer checking a result rather than error instance because it needs only one block.

```
  if (!g_span_async (...)) {
    // treat error state.
  }

  // normal procedure.
```

Reply:
Sure. I would include this in the next PR